### PR TITLE
App Menu and Settings > General Changes

### PR DESCRIPTION
**Pull request by hamishwillee**
This updates 
1. General > Settings
2. Fixes up changes from removal of the app menu: 
   - remove widgets for: onboard files, custom control, HIL settings
   - move mavlink inspector and update to have charts.
   - move replay flight data and link to settings.
   - remove mavlink analyser

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
> How do we access replay and the other missing features?

* You can turn on the log replay toolbar from Settings/Fly View
* Mavlink Inspector and Analyze are now under the Analyze View
* Custom Command is no longer supported

Not sure about Onboard Files I thought I had removed that ages ago from the build. Mainly because it doesn't really work. It should stay gone as well.

**Comment by hamishwillee** [comment]
Cool. Are these features going to be available in all builds - e.g. both daily and stable?

**Comment by hamishwillee** [comment]
Also, any thoughts about MAVLink Analyze - not where I think you said in todays daily build

image

**Comment by hamishwillee** [comment]
FMI, would the Replay Data not be better off in Analyze than in Fly? The bar appears on all views, and watching it on the map is only one part of it.

**Comment by DonLakeFlyer** [comment]
> Also, any thoughts about MAVLink Analyze - not where I think you said in todays daily build
> 
> image

Very strange. @dogmaphobic What happened to charting?

**Comment by DonLakeFlyer** [comment]
> FMI, would the Replay Data not be better off in Analyze than in Fly?

It's a setting, so it's on the settings pages Not quite sure what you mean?

**Comment by dogmaphobic** [comment]
> Very strange. @dogmaphobic What happened to charting?

It's part of the MAVLink Inspector:

Screen Shot 2020-02-11 at 12 48 29 PM

**Comment by DonLakeFlyer** [comment]
Ah, crap right forgot that. I wonder if the button name should change somehow then. Worried too many folks will think it's just gone.

**Comment by DonLakeFlyer** [comment]
Other possibility is to change the icon to a chart looking thing as well.

**Comment by dogmaphobic** [comment]
> Other possibility is to change the icon to a chart looking thing as well.

Yup, that's the last disruptive. Let me look into it.

**Comment by dogmaphobic** [comment]
https://github.com/mavlink/qgroundcontrol/pull/8311

**Comment by hamishwillee** [comment]
>> FMI, would the Replay Data not be better off in Analyze than in Fly?

> It's a setting, so it's on the settings pages Not quite sure what you mean?

The replay data setting is in **General > Fly View** and I am proposing it is moved to a new section **General > Analyze View**. 

It is a minor point, but I don't think I'd expect to find this setting under fly view - more under analyze. When I do the docs I'd also rather move the feature to analyze than fly view sections.

**Comment by hamishwillee** [comment]
>> Other possibility is to change the icon to a chart looking thing as well.

> Yup, that's the least disruptive. Let me look into it.

@dogmaphobic Works for me. I'm hoping you guys include the move of the feature in the release note too.

**Comment by hamishwillee** [comment]
@dogmaphobic Shouldn't the values in mavlink inspector show the last received? Here you can see we received one autopilot version message, but no values shown

image

File: en/SettingsView/General.md [hunk 1]
```diff
@@ -27,51 +27,71 @@ This section defines a number of miscellaneous settings, related to (non exhaust
...
-- **Save telemetry log even if vehicle was not armed**: Logs when a vehicle connects to *QGroundControl*. 
-  Stops logging when the last vehicle disconnects.
-- **Use preflight checklist**: Enable pre-flight checklist.
-- **Virtual Joystick**: Enable [virtual joysticks](../SettingsView/VirtualJoystick.md) (PX4 only)
+- **Map Provider**: Google | Mapbox | Bing | Airmap | VWorld | Eniro | Statkart
+- **Map Type**: Road | Hybrid | Satellite
+- **Stream GCS Position**: Never | Always | When in Follow Me Flight Mode.
+- **UI Scaling**: UI scale percentage (affects fonts, icons, button sizes, layout etc.) 
+- **Mute all audio output**: Turns off all audio output.
+- **Check for Internet Connection**: TBD
```
Comment (hamishwillee) [inline en/SettingsView/General.md]: @DonLakeFlyer What is check for internet connection for/how does it work?

File: en/SettingsView/General.md [hunk 2]
```diff
@@ -27,51 +27,71 @@ This section defines a number of miscellaneous settings, related to (non exhaust
...
+![Fly View Settings](../../assets/settings/general/fly_view.jpg)
+
+The settings are:
+- **Use Preflight Checklist**: Enable pre-flight checklist in Fly toolbar.
+- **Enforce Preflight Checklist**: Checklist completion is a pre-condition for arming.
+- **Keep Map Centered on Vehicle**: Forces map to center on the currently selected vehicle.
+- **Show Telemetry Log Replay Status Bar**: Display status bar for [Replaying Flight Data](../FlyView/replay_flight_data.md).
+- **Virtual Joystick**: Enable [virtual joysticks](../SettingsView/VirtualJoystick.md) (PX4 only)
+- **Use Vertical Instrument Panel**: Align instrument panel vertically rather than horizontally (default).
+- **Show additional heading indicators on Compass**: TBD
```
Comment (hamishwillee) [inline en/SettingsView/General.md]: @DonLakeFlyer I tried setting additional heading indicators but can't tell what they do. Can you outline their purpose. Ideally with screenshot.

File: en/SettingsView/General.md [hunk 3]
```diff
@@ -119,14 +139,61 @@ Use the *NMEA GPS Device* drop-down selector to manually select the GPS device a
...
+
+
+
+
+## ADBS Server {#adbs_server}
+
+![ADBS_Server Settings](../../assets/settings/general/adbs_server.jpg)
+
+The settings are:
+- **Connect to ADBS SBS server**: TBD
```
Comment (hamishwillee) [inline en/SettingsView/General.md]: @DonLakeFlyer Can you tell me a bit more about what this is for? How it works? When it should be set/not set?

File: en/analyze_view/mavlink_inspector.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# MAVLink Inspector Widget
```
Comment (hamishwillee) [inline en/analyze_view/mavlink_inspector.md]: @dogmaphobic This is update to MAVLink inspector docs. Can you please sanity check.

**Comment by hamishwillee** [comment]
OK, this is nearly ready (modulo answer to query https://github.com/mavlink/qgc-user-guide/pull/237#issuecomment-584920432 and your reviews).

This is for a new stable version so perhaps we hold off pushing it in until that is about to go live?

File: en/SettingsView/General.md (hunk 3 above)
Comment (DonLakeFlyer) [reply en/SettingsView/General.md]: It works pretty close to how MP works: https://discuss.ardupilot.org/t/ads-b-mission-planner-and-flightaware-ads-b-dongle/21362. Not a super easy thing to set up though.

File: en/SettingsView/General.md (hunk 1 above)
Comment (DonLakeFlyer) [reply en/SettingsView/General.md]: That would be @dogmaphobic. First I'm seeing that one.

Comment (dogmaphobic) [reply en/SettingsView/General.md]: QGC needs Internet for downloading map tiles. If you are not connected to the Internet, the map tile engine will keep trying to download and failing. To stop it from doing that, we check to see if you actually have internet. If you don't the map tile engine is disabled. 

The problem is that the way we test for Internet doesn't work in China. It always returns negative (telling you don't have internet) and therefore, maps are disabled. This was a way to allow Chinese to use maps. If this setting is set, we skip the tests and assume Internet is present.

File: en/SettingsView/General.md (hunk 2 above)
Comment (DonLakeFlyer) [reply en/SettingsView/General.md]: Screen Shot 2020-02-12 at 9 50 53 AM

* Blue Array - Course Over Ground
* White House - Direction back to home
* Green Line - Direction to next waypoint

If you run Sitl, takeoff and do a go to location you can get these to move around

**Comment by DonLakeFlyer** [comment]
All the "1"s are component id I believe.

File: en/SettingsView/General.md (hunk 1 above)
Comment (DonLakeFlyer) [reply en/SettingsView/General.md]: Change the name to something which talks about map download as well?

**Comment by dogmaphobic** [comment]
> Shouldn't the values in mavlink inspector show the last received? Here you can see we received one autopilot version message, but no values shown

Not sure what's going on there. I need to check.

> Further, all of the messages have a "1" displayed. What's that? Component id?

Correct. On a normal system, you would see various component IDs. That is, you might see several "Heartbeat" messages, each for a different component ID. You can also filter the messages by component ID.

File: en/SettingsView/General.md (hunk 3 above)
Comment (hamishwillee) [reply en/SettingsView/General.md]: Sorry, that is not precise enough. 

So my assumption is that this allows you to connect to an ADBS server "somewhere" in order to get ADBS information. You select connect, enter the URL/port of the server, and QGC shows you info it has from the server about vehicles.
1. Is that about right?
2. Are there public servers or do people have to set up their own?
3. What does QGC display for each vehicle it detects?
2. What gets displayed by QGC if there is a sever

File: en/SettingsView/General.md (hunk 1 above)
Comment (hamishwillee) [reply en/SettingsView/General.md]: Maybe "Map Download Internet Check"?

I have changed to 

> - **Check for Internet Connection**: Uncheck to allow maps to be used in China/places where map tile downloads are likely to fail (stops the map-tile engine continually rechecking for an Internet connection).

**Comment by DonLakeFlyer** [comment]
Any else needed before this get merged?

**Comment by hamishwillee** [comment]
@DonLakeFlyer This was waiting on a sanity check, and the update the changes reflect the current stable release).

**Comment by DonLakeFlyer** [comment]
Looks fine to me

**Comment by hamishwillee** [comment]
Thanks. You made the 4.0 release public, so I figure this can be merged.

**Pull request merged**